src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -328,6 +331,46 @@ take n (Fold fstep finitial fextract) = Parser step initial extract
 
     extract (Tuple' _ r) = fextract r
 
+--
+-- | See 'Streamly.Internal.Data.Parser.takeBetween'.
+--
+-- /Internal/
+--
+{-# INLINE takeBetween #-}
+takeBetween :: MonadCatch m => Int -> Int -> Fold m a b -> Parser m a b
+takeBetween low high (Fold fstep finitial fextract) = 
+    
+    Parser step initial extract
+
+    where
+    
+    initial = Tuple' 0 <$> finitial
+
+    step (Tuple' numTaken s) a =
+        if high == 0
+        then Done 1 <$> fextract s
+        else
```

Comment:
Is this the correct behavior? If my understanding is correct: `takeBetween low high` takes all the values between `low` and `high` right? In this case, it takes everything below `low` as well.
`takeBetween 2 3 toList [0, 1, 2, 3, 4]` should be `[2]`.
According to the implementation, it is `[0, 1, 2]`

Reply:
Ignore this, I seem to have misunderstood.